src/index.ts

```diff
@@ -179,9 +179,15 @@ export async function init(config: IClientConfig): Promise<IEppoClient> {
     EppoJSClient.instance.stopPolling();
     // Set up assignment logger and cache
     EppoJSClient.instance.setLogger(config.assignmentLogger);
+
+    // Note that we use the first 8 characters of the API key to create per-API key persistent storages
+    const storageKeySuffix = config.apiKey.replace(/\W/g, '').substring(0, 8);
+
     // default behavior is to use a LocalStorage-based assignment cache.
     // this can be overridden after initialization.
-    EppoJSClient.instance.useCustomAssignmentCache(new LocalStorageAssignmentCache());
+    EppoJSClient.instance.useCustomAssignmentCache(
+      new LocalStorageAssignmentCache(storageKeySuffix),
+    );
```

Comment:
FYI - a factory is needed here as well https://linear.app/eppo/issue/FF-2113/create-chromestorage-assignment-cache